ui/components/ui/delineator/delineator.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useState } from 'react';
+import classnames from 'classnames';
+import {
+  BorderRadius,
+  AlignItems,
+  BackgroundColor,
+  IconColor,
+  Display,
+  JustifyContent,
+  FlexDirection,
+} from '../../../helpers/constants/design-system';
+import {
+  AvatarIcon,
+  Icon,
+  Box,
+  IconName,
+  IconSize,
+} from '../../component-library';
+import Preloader from '../icon/preloader/preloader-icon.component';
+import { DelineatorProps } from './delineator.types';
+import { getIconPropsByType, overrideTextComponentColorByType } from './utils';
+
+/**
+ * Renders the icon on the right based on the loading state.
+ *
+ * @param options - Options bag
+ * @param options.isLoading - Whether the delineator is in a loading state.
+ * @param options.isExpanded - Whether the delineator is expanded.
+ * @returns Either a loading spinner or an arrow icon.
+ */
+const ExpandableIcon = ({
+  isLoading,
+  isExpanded,
+}: {
+  isLoading: boolean;
+  isExpanded: boolean;
+}) => {
+  if (isLoading) {
```

Comment:
If these two boolean properties are mutually exclusive, meaning this component is really just two components, then would it be cleaner to define them as separate components  such as `Loader` and `ExpandableIcon` and then render them with their conditions in the parent such as:

```
{isLoading && <Loader/>}
{!isLoading && <ExpandableIcon isExpanded={isExpanded}/>}
```

Reply:
Make sense, fixed this